common/desktop-exports

```diff
@@ -103,7 +103,10 @@ mkdir -p $XDG_CONFIG_HOME
 append_dir LOCPATH $RUNTIME/usr/lib/locale
 
 # Run xdg-user-dirs-update
-if [ `which xdg-user-dirs-update` ]; then
+if [ -f $REALHOME/.config/user-dirs.dirs ] && [ -f $REALHOME/.config/user-dirs.locale ]; then
```

Comment:
This is not enough to check if we have access to such files as per https://bugs.launchpad.net/snappy/+bug/1748510
You should use something that uses `open()`

One option is `head -c0` (which just opens and closes), so maybe add an utility function such as:

```bash
function can_open_file() {
  exit ! head -c0 "$1" &> /dev/null;
}
```